bpf/go_nethttp.c

```diff
@@ -61,6 +61,8 @@ int uprobe_ServeHTTP(struct pt_regs *ctx) {
 
     if (req) {
         server_trace_parent(goroutine_addr, &invocation.tp, (void*)(req + req_header_ptr_pos));
+        // TODO: if context propagation is supported, overwrite the header value in the map with the 
+        // new span context and the same thread id.
```

Comment:
I don't see it as incomplete in terms of Beyla instrumentation, but it would be nice to make Beyla work well for instrumented applications with the Go SDK. As it stands now, Beyla will correctly propagate the context, but if there's SDK instrumentation as well, the two spans will look parallel, just as it was before. If we overwrite the header value, we'll manage to nest them. So it's an extension to the feature to make it work with auto and manual instrumentation.